### DISCUSSION TITLE
Big Refactor

### DISCUSSION THREAD

**Discussion by echavet** [echavet/MitsubishiCN105ESPHome#30] [Announcements]
## What's Changed
* Optimistic refactor by @echavet in https://github.com/echavet/MitsubishiCN105ESPHome/pull/28
* addition of the new RQST_PKT_STANDBY step in the loop cycle by @echavet in https://github.com/echavet/MitsubishiCN105ESPHome/pull/29

**Full Changelog**: https://github.com/echavet/MitsubishiCN105ESPHome/compare/v1.1.3...v1.2

<hr /><em>This discussion was created from the release <a href='https://github.com/echavet/MitsubishiCN105ESPHome/releases/tag/v1.2'>Big Refactor</a>.</em>

**Comment by disruptivepatternmaterial** [comment]
I pushed this from main to all my controllers and they seem to be working fine. Will keep an eye on them and report anythings I see.

**Reply by disruptivepatternmaterial** [reply]
So far so good! Not single issue. All running esp32s3 using esp-idf.

**Reply by echavet** [reply]
Regularly check that the state of the Home Assistant component remains synchronized with the air conditioning.

It happened to me twice that the component the heatpump did not get the update packet. But the component is now optimistic.

I've opened an issue: #40 

I've made some little changes again to force the sync at reception of an update packet.
